glue/core/tests/test_roi.py

```diff
@@ -340,6 +341,36 @@ def test_str(self):
         assert type(str(self.roi)) == str
 
 
+class TestCategorical(object):
+
+    def setup_method(self, method):
+        self.roi = CategoricalRoi()
+
+    def test_defined(self):
+
+        nroi = CategoricalRoi(categories=['a', 'b', 'c'])
+        assert nroi.defined()
+        nroi.reset()
+        assert not nroi.defined()
+
+    def test_loads_from_components(self):
+
+        comp = CategoricalComponent(np.array(['a', 'a', 'b']))
+        self.roi.update_categories(comp)
+
+        np.testing.assert_array_equal(self.roi.categories,
+                                      np.array(['a', 'b']))
+
+    def test_applies_components(self):
+
+        comp = CategoricalComponent(np.array(['a', 'b', 'c']))
+        self.roi.update_categories(CategoricalComponent(np.array(['a', 'b'])))
+        contained = self.roi.contains(comp, None)
+        np.testing.assert_array_equal(contained,
+                                      np.array([True, True, False]))
+
+
+
```

Comment:
Correct. In the CategoricalRoi code proper I have some logic to deal with inputs of `list`, `np.array`, `pd.Series` and `CategoricalComponents`and transparently deal with all variants.

With the components I access their underlying `.categorical_data` to get the actual data instead of the digitized versions. This way it will be comparable across different `Data` objects, something that didn't really work in the previous iterations.